### DISCUSSION TITLE
useUnmountedRef 的一个疑问

### DISCUSSION THREAD

**Discussion by gaowujie2004** [alibaba/hooks#1546] [Q&A] [answered]
https://github.com/alibaba/hooks/blob/4e2a133264fa13c1b4098a8fa5ae14da3bbcbf43/packages/hooks/src/useUnmountedRef/index.ts#L4

```
import { useEffect, useRef } from 'react';

const useUnmountedRef = () => {
  const unmountedRef = useRef(false);
  useEffect(() => {
    unmountedRef.current = false;
    return () => {
      unmountedRef.current = true;
    };
  }, []);
  return unmountedRef;
};

export default useUnmountedRef;
```

作为小白的我，我觉得 这行显的有些多余了。可以解答一下我的疑问吗，如果删除这行，什么样的场景下会有问题，我将 感激不尽！
unmountedRef.current = false;

**Comment by brickspert** [comment, marked as answer]
可以看一下这个文档：https://ahooks.js.org/zh-CN/guide/blog/hmr

**Reply by gaowujie2004** [reply]
感谢您的回复，我的问题解决了。非常感谢！
